app/src/main/java/ch/epfl/sdp/appart/database/firestoreservicehelpers/FirestoreAdHelper.java

```diff
@@ -149,7 +155,19 @@ private void getAndCheckPartialAd(CompletableFuture<Ad.AdBuilder> result,
                         .withDescription((String) documentSnapshot.get(AdLayout.DESCRIPTION))
                         .hasVRTour((boolean) documentSnapshot.get(AdLayout.VR_TOUR));
 
-                result.complete(builder);
+                db.collection(FirebaseLayout.USERS_DIRECTORY).document((String) documentSnapshot.get(AdLayout.ADVERTISER_ID)).get().addOnCompleteListener(
+                        userTask -> {
+                            if (userTask.isSuccessful()) {
+                                Map<String, Object> data = userTask.getResult().getData();
+                                String advertiserName = userSerializer.deserialize((String) documentSnapshot.get(AdLayout.ADVERTISER_ID), data).getName();
+                                builder.withAdvertiserName(advertiserName);
+                                result.complete(builder);
+                            } else {
```

Comment:
It feels strange to fetch the whole user to use just the name, there should be a way to retrieve just the name once we have the reference.

Reply:
Very good point, changed it! it is fixed now.